neuro_san/internals/journals/progress_journal.py

```diff
@@ -0,0 +1,53 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from typing import Any
+from typing import Dict
+
+from neuro_san.interfaces.agent_network_progress_reporter import AgentNetworkProgressReporter
+from neuro_san.internals.journals.journal import Journal
+from neuro_san.internals.messages.agent_progress_message import AgentProgressMessage
+
+
+class ProgressJournal(AgentNetworkProgressReporter):
+    """
+    An implementation of the AgentNetworkProgressReporter interface for a CodedTool to be able
+    to journal AgentProgressMessages.
+    """
```

Comment:
New class to implement the AgentNetworkProgressReporter interface seen by the CodedTools.
This is created by the neuro-san infrastructure in AbstractClassActivation